packages/block-editor/src/components/gradients/use-gradient.js

```diff
@@ -59,22 +59,22 @@ export function getGradientSlugByValue( gradients, value ) {
 	return gradient && gradient.slug;
 }
 
-const EMPTY_OBJECT = {};
-
 export function __experimentalUseGradient( {
 	gradientAttribute = 'gradient',
 	customGradientAttribute = 'customGradient',
 } = {} ) {
 	const { clientId } = useBlockEditContext();
 
-	const gradientsPerOrigin = useSetting( 'color.gradients' ) || EMPTY_OBJECT;
+	const userGradientPalette = useSetting( 'color.gradients.custom' );
```

Comment:
Can you explain a bit why `useSetting( 'color.gradients' )` wouldn't work? Is it because we want to remove support for retrieval of top level settings(which this is not) and a combination of custom handling for some settings (origin priority and deprecated fallbacks? Do we have a problem/different handling for other settings?

I saw a lot of connected PRs about this but I don't have much context in this area and would like to understand it better.